apis/net/v1alpha1/tunnelEndpointClient.go

```diff
@@ -58,5 +59,5 @@ func Keyer(obj runtime.Object) (string, error) {
 		return "", errors.New("cannot cast received object to TunnelEndpoint")
 	}
 
-	return tunEnd.Name, nil
+	return fmt.Sprintf("%s/%s", tunEnd.Namespace, tunEnd.Name), nil
```

Comment:
You could use the cache.MetaNamespaceKeyFunc, which automatically handles this operation.